lib/rules/valid-typeof.js

```diff
@@ -74,6 +93,8 @@ module.exports = {
                             }
                         } else if (requireStringLiterals && !isTypeofExpression(sibling)) {
                             context.report({ node: sibling, messageId: "notString" });
+                        } else if (sibling.type === "Identifier" && sibling.name === "undefined" && isReferenceToGlobalVariable(sibling)) {
+                            context.report({ node: sibling, messageId: "invalidValue" });
```

Comment:
Besides reporting the error, I think we can provide a fix.

Reply:
How can you be sure of the user’s intention?

Reply:
For `typeof foo === symbol`, no because `symbol` may have been initialized with `"symbol"`. But for `typeof foo === undefined`, yes, that's definitely a typo.

Reply:
undefined is a variable too, and could be initialized to anything.

Either way, this seems safer as a suggestion and not an autofix.

Reply:
ok, an autofix is too aggressive. it comes to my mind that some crazy guy might write the following code.
```js
var env = {
   undefined: 'number'
};

with (env) {
   if (typeof 1 === undefined) {
      console.log('ooops')
   }
}
```

Reply:
We don't auto-fix code if that would change its behavior, even if the new behavior is most likely the intended one.

[provoding suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) sounds good in this case.